### PR TITLE
METRON-2013: The bro plugin docker script's topic name should be configurable

### PR DESCRIPTION
**Pull request by JonZeolla**
## Contributor Comments
This was spurred by the METRON-2003 [review process](apache/metron-bro-plugin-kafka/pull/26).

I was a bit conflicted on if passing `--kafka-topic` to `./run_end_to_end.sh` should also modify `in_docker_scripts/configure_bro_plugin.sh` to use the provided topic name (instead of the hard coded "bro").  Looking for feedback - I could see it either way but haven't thought too critically about it yet.

### Testing
1. Use `./run_end_to_end.sh` with no `--kafka-topic` param, or with `--kafka-topic=anythingbutbro`.  If you set `--kafka-topic` to anything but bro, modify `in_docker_scripts/configure_bro_plugin.sh` appropriately.  The outputs should be the same as before this PR.
1. Confirm that when you provided the kafka topic, such as by looking at that kafka container's /data folder.
```
./scripts/docker_execute_shell.sh --container-name=kafka
ls /data | grep INSERT_KAFKA_TOPIC_NAME_HERE
```

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron's Bro kafka writer plugin.

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [X] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [X] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [X] Have you included steps or a guide to how the change may be verified and tested manually?
- [X] Have you ensured that the full suite of tests and checks have been executed via:
  ```
  bro-pkg test $GITHUB_USERNAME/metron-bro-plugin-kafka --version $BRANCH
  ```
- [X] Have you written or updated unit tests and or integration tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] Have you verified the basic functionality of the build by building and running locally with Apache Metron's [Vagrant full-dev environment](https://github.com/apache/metron/tree/master/metron-deployment/development/centos6) or the equivalent?

### CODE REVIEW

**Comment by JonZeolla** [comment]
This probably needs a closer look at the docs.  I did take a first stab at that, but could look more later this week to confirm I didn't miss anything.

File: docker/scripts/docker_run_get_offset_kafka.sh [hunk 1]
```diff
@@ -24,19 +24,21 @@ set -E # errtrap
...
-# all of the messages available
+# Runs a kafka container with the console consumer for the provided topic.  The
+# consumer should quit when it has read all of the messages available.
 #
 
 function help {
   echo " "
   echo "usage: ${0}"
   echo "    --network-name                  [OPTIONAL] The Docker network name. Default: bro-network"
+  echo "    --kafka-topic                   [OPTIONAL] The kafka topic to theee the offset from. Default: bro"
```
Comment (JonZeolla) [inline docker/scripts/docker_run_get_offset_kafka.sh]: Typo

File: docker/README.md [hunk 2]
```diff
@@ -70,9 +70,9 @@ testing scripts to be added to a pull request, and subsequently to a test suite.
 ├── docker_execute_process_data_file.sh
 ├── docker_execute_shell.sh
 ├── docker_run_bro_container.sh
-├── docker_run_consume_bro_kafka.sh
-├── docker_run_create_bro_topic_in_kafka.sh
-├── docker_run_get_offset_bro_topic_in_kafka.sh
+├── docker_run_consume_kafka.sh
+├── docker_run_get_offset_kafka.sh
```
Comment (JonZeolla) [inline docker/README.md]: Reorder

File: docker/README.md [hunk 3]
```diff
@@ -140,22 +140,25 @@ testing scripts to be added to a pull request, and subsequently to a test suite.
...
   --offset                       [OPTIONAL] The kafka offset. Default: -1
+  --kafka-topic                  [OPTIONAL] The kafka topic to consume from. Default: bro
   ```
-- `docker_run_get_offset_bro_kafka.sh`: Runs an instance of the kafka container and gets the current offset for the bro topic
+- `docker_run_get_offset_kafka.sh`: Runs an instance of the kafka container and gets the current offset for the specified topic
   ###### Parameters
   ```bash
   --network-name                 [OPTIONAL] The Docker network name. Default: bro-network
   --offset                       [OPTIONAL] The kafka offset. Default: -1
+  --kafka-topic                  [OPTIONAL] The kafka topic to theee the offset from. Default: bro
```
Comment (JonZeolla) [inline docker/README.md]: Typo

**Comment by ottobackwards** [comment]
I don't understand what you are doing here with the consume and the get offset scripts.
The issue is that we will support a 'matrix' of topic options. How does us move this towards that?

**Comment by JonZeolla** [comment]
Something got messed up when I was renaming files, that was not intended.  Let me fix it up

**Comment by JonZeolla** [comment]
The goal is to allow a configurable topic name, to simplify certain tests like seen in METRON-2003.  I think in the future we will be able to build on this for multiple topics as well, but I'm leaving that out for now.

**Comment by JonZeolla** [comment]
Okay, after a little bit of confusion I've reset things back to where I started.  There was no issue with how this was, but git _was_ showing the wrong file renames because the two files were so simple, which resulted in the "files changed" page on github also being misleading.

**Comment by JonZeolla** [comment]
Deconflicted, this should be good to go.

**Comment by ottobackwards** [comment]
+1

**Pull request closed without merging**